### PR TITLE
chore: drop usage of jquery in `install` and `update` interfaces

### LINKED ISSUE #3457: Migrate update and install pages away from legacy external jQuery
https://github.com/flarum/framework/blob/main/framework/core/views/install/update.php#L20

https://github.com/flarum/framework/blob/main/framework/core/views/install/install.php#L69

### PR DESCRIPTION
**Pull request by SychO9**
**Fixes #3457**

**Changes proposed in this pull request:**
* Just converts the JS code to vanilla JS.
* Tesed locally.

**Necessity**

- [x] Has the problem that is being solved here been clearly explained?
- [x] If applicable, have various options for solving this problem been considered?
- [x] For core PRs, does this need to be in core, or could it be in an extension?
- [x] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

File: framework/core/views/install/install.php [hunk 1]
```diff
@@ -66,29 +66,38 @@
   </div>
 </form>
 
-<script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
 <script>
-$(function() {
-  $('form :input:first').select();
+document.querySelector('form input').select();
```
Comment (askvortsov1) [inline framework/core/views/install/install.php]: Should this be in an `onload`?

File: framework/core/views/install/install.php [hunk 2]
```diff
@@ -66,29 +66,38 @@
...
+        response.text().then(errorMessage => {
+          var error = document.querySelector('#error');
+          error.style.display = 'block';
+          error.textContent = 'Something went wrong:\n\n' + errorMessage;
+          button.disabled = false;
+          button.textContent = 'Install Flarum';
+        });
+      }
+    })
+    .catch(error => {
```
Comment (askvortsov1) [inline framework/core/views/install/install.php]: Why are we catching the error to immediately rethrow it?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: framework/core/views/install/install.php (hunk 2 above)
Comment (SychO9) [reply framework/core/views/install/install.php]: woups 🙈 was debugging

File: framework/core/views/install/update.php [hunk 3]
```diff
@@ -18,37 +18,36 @@
...
-            error.textContent = 'Something went wrong:\n\n' + errorMessage;
-            button.disabled = false;
-            button.textContent = 'Update Flarum';
-          });
-        }
-      })
-      .catch(error => {
-        console.error('Error:', error);
-      });
+  window.onload = function() {
```
Comment (askvortsov1) [inline framework/core/views/install/update.php]: https://stackoverflow.com/a/37401912 might be a better fit, actually

Comment (SychO9) [reply framework/core/views/install/update.php]: let me just switch back to jQuery 🤣

Comment (askvortsov1) [reply framework/core/views/install/update.php]: Aaaaaaaaa noooooo!!!!!!!

**Review by askvortsov1 — APPROVED** [review]

**Comment by matteocontrini** [comment]
A bit late here, but I think fetch Promise rejections should be handled too. Otherwise network errors are ignored (they weren't, in the previous jQuery implementation).

**Pull request merged**